ExtraEnemyCustomization/Utils/Vector3Converter.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using UnityEngine;
+
+namespace EECustom.Utils
+{
+    public class Vector3Converter : JsonConverter<Vector3>
+    {
+        public override bool HandleNull => false;
+
+        public override Vector3 Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var vector = new Vector3();
+
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.StartObject:
+                    while (reader.Read())
+                    {
+                        if (reader.TokenType == JsonTokenType.EndObject)
+                            return vector;
+
+                        if (reader.TokenType != JsonTokenType.PropertyName)
+                            throw new JsonException("Expected PropertyName token");
+
+                        var propName = reader.GetString();
+                        reader.Read();
+
+                        switch (propName.ToLower())
+                        {
+                            case "x":
+                                vector.x = reader.GetSingle();
+                                break;
+
+                            case "y":
+                                vector.y = reader.GetSingle();
+                                break;
+
+                            case "z":
+                                vector.z = reader.GetSingle();
+                                break;
+                        }
+                    }
+                    throw new JsonException("Expected EndObject token");
+
+                case JsonTokenType.String:
+                    var strValue = reader.GetString().Trim();
+                    if (TryParseVector3(strValue, out vector))
+                    {
+                        return vector;
+                    }
+                    throw new JsonException($"Vector3 format is not right: {strValue}");
+
+                default:
+                    throw new JsonException($"Vector3Json type: {reader.TokenType} is not implemented!");
+            }
+        }
+
+        private bool TryParseVector3(string input, out Vector3 vector)
+        {
+            input = input.Trim().Trim('(', ')');
```

Comment:
It might be better to strip non numeric characters with a regular expression. Something like `(?:[^0-9])([0-9.]+)`